code/game/objects/items.dm

```diff
@@ -1420,21 +1424,24 @@ var/global/list/image/blood_overlays = list()
 	if (!bodypart)//no bodypart specified? that should mean we're being held.
 		bodypart = HANDS
 	//secondly, do they happen to carry contact-spreading viruses themselves?
-	var/list/contact_diseases = filter_disease_by_spread(M.virus2,required = SPREAD_CONTACT)
+	var/list/contact_diseases = filter_disease_by_spread(M.virus2,required = SPREAD_CONTACT | SPREAD_COLONY)
 	if (contact_diseases?.len)
 		//if so are their hands protected?
-		if (!M.check_contact_sterility(bodypart))
-			for (var/ID in contact_diseases)
-				var/datum/disease2/disease/D = contact_diseases[ID]
+		var/block = M.check_contact_sterility(bodypart)
+		for (var/ID in contact_diseases)
+			var/datum/disease2/disease/D = contact_diseases[ID]
+			if(attempt_colony(src,D,"from being touched by [M]"))
```

Comment:
I'm not sure what this is supposed to do. Wouldn't
```dm
if (!attempt_colony(...) && !block)
     infect_disease2(...)
```
Work similarly, and be clearer ?

Reply:
attempt_colony() doesn't care about block in a positive OR negative way. It has its own check to see if the target is a suitable colony (which has nothing to do with coverage or sterility - instead it wants to know it an atmos-resistant item is being worn in the suit slot)

The reason for this is so that it won't try infecting a second time if it has already successfully infected via COLONY spread, it won't try contact spread.